src/machine/state/mod.rs

```diff
@@ -16,6 +17,49 @@ pub struct State {
 
 impl State {
     pub fn new() -> Self {
-        unimplemented!();
+        State {
+            // for the moment start execution at address 0
+            pc: 0,
+
+            sr: SRegisters::new(),
+            gpr: GPRegisters::new(),
+
+            // for the moment create 1MB of memory
+            mem: Memory::with_capacity(0x100000)
+        }
+    }
+
+    /// Display the state partially on the command line for testing purposes
+    pub fn debug_output(&self) {
```

Comment:
Mhmm... Do we really need to pass parameters? I don't think so. And even if, I think we can still find a good way then.